test/lifecycle_module_test.dart

```diff
@@ -720,57 +906,153 @@ void main() {
-            'didUnloadChildModule',
-            'onUnload',
-            'didUnload'
-          ]));
-      expect(
-          childModule.eventList,
-          equals([
-            'onShouldUnload',
-            'onShouldUnload',
-            'willUnload',
-            'onUnload',
-            'didUnload'
-          ]));
+    group('unload', () {
+      test('should unload child modules', () async {
+        await parentModule.loadChildModule(childModule);
+        parentModule.eventList.clear();
+        childModule.eventList.clear();
+        await parentModule.unload();
+        expect(
+            parentModule.eventList,
+            equals([
+              'onShouldUnload',
+              'willUnload',
+              'onWillUnloadChildModule',
+              'willUnloadChildModule',
+              'onDidUnloadChildModule',
+              'didUnloadChildModule',
+              'onUnload',
+              'didUnload'
+            ]));
+        expect(
+            childModule.eventList,
+            equals([
+              'onShouldUnload',
+              'onShouldUnload',
+              'willUnload',
+              'onUnload',
+              'didUnload'
+            ]));
+      });
+
+      group('with an child with an onUnload that throws', () {
+        setUp(() async {
+          await parentModule.loadChildModule(childModule);
+          childModule.onUnloadError = testError;
+        });
+
+        test(
+            'should return the child error',
+            () => expect(parentModule.unload(),
+                throwsA(same(childModule.onUnloadError))));
+
+        test('should add that error to didUnload stream', () {
+          parentModule.didUnload.listen((LifecycleModule _) {},
+              onError: expectAsync1((Error error) =>
+                  expect(error, same(childModule.onUnloadError))));
+          expect(
+              parentModule.unload(), throwsA(same(childModule.onUnloadError)));
+        });
+      });
+
+      group('with a parent with an onWillUnloadChildModule that throws', () {
+        setUp(() async {
+          await parentModule.loadChildModule(childModule);
+          parentModule.onWillUnloadChildModuleError = testError;
+        });
+
+        test('should add that error to onWillUnloadChildModule stream', () {
+          parentModule.willUnloadChildModule.listen((LifecycleModule _) {},
+              onError: expectAsync1((Error error) => expect(
+                  error, same(parentModule.onWillUnloadChildModuleError))));
+          parentModule.unload();
+        });
+      });
+
+      group('with a parent with an onDidUnloadChildModule that throws', () {
+        setUp(() async {
+          await parentModule.loadChildModule(childModule);
+          parentModule.onDidUnloadChildModuleError = testError;
+        });
+
+        test('should add that error to onWillUnloadChildModule stream', () {
+          parentModule.didUnloadChildModule.listen((LifecycleModule _) {},
```

Comment:
Is there a mismatch between the test description and test contents? 'onWillUnloadChildModule' vs 'didUnloadChildModule' ?